File: app/src/main/java/com/chris/eban/data/EventMapper.java
package com.chris.eban.data;

import com.chris.eban.domain.entity.DMEventListItem;

import java.util.ArrayList;
import java.util.List;

class EventMapper {
    List<DMEventListItem> map(List<Event> events) {
        ArrayList<DMEventListItem> dmEventListItems = new ArrayList<>();
        for (Event event : events) {
            dmEventListItems.add(map(event));
        }
        return dmEventListItems;
    }

    public DMEventListItem map(Event event) {
        DMEventListItem dmEventListItem = new DMEventListItem();
        dmEventListItem.id = event.id;
        dmEventListItem.title = event.title;
        dmEventListItem.content = event.content;
        return dmEventListItem;
    }

    Event map(DMEventListItem item) {
        Event event = new Event();
        event.id = item.id;
        event.content = item.content;
        event.title = item.title;
        return event;
    }
}

File: app/src/main/java/com/chris/eban/data/EventMapper.kt
package com.chris.eban.data

import com.chris.eban.domain.entity.DMEventListItem
import java.util.*

internal class EventMapper {

    fun map(events: List<DataEvent>): List<DMEventListItem> {

        val dmEventListItems = ArrayList<DMEventListItem>()
        for (event in events) {
            dmEventListItems.add(map(event))
        }
        return dmEventListItems
    }

    fun map(event: DataEvent): DMEventListItem {

        val dmEventListItem = DMEventListItem()
        dmEventListItem.id = event.id
        dmEventListItem.title = event.title
        dmEventListItem.content = event.content

        dmEventListItem.createDate = event.createTime
        dmEventListItem.updateDate = event.updateTime
        return dmEventListItem
    }

    fun map(item: DMEventListItem): DataEvent {
        return DataEvent(item.id, item.title, item.content, item.createDate, item.updateDate)
    }
}
